_docs/README.md

```diff
@@ -0,0 +1,275 @@
+1. Embed code snippets from actual source files in documentation. Most READMEs have code copy/pasted into them and that
+   code often has syntax errors or goes out of date. Now you can keep those code examples in normal source files which
+   are built & tested, and embed parts of those files dynamically in your docs.
+
+Note: the README for this project is generated using boilerplate! Check out the templates for it in the
+[_docs folder](/_docs) and the build job configuration in [circle.yml](/circle.yml).
+
+## Example: creating a new template
+
+Create a folder called `website-boilerplate` and put a file called `boilerplate.yml` in it:
+
+```json
+{{snippet "../examples/website/boilerplate.yml"}}
+```
+
+This file defines 3 variables: `Title`, `WelcomeText`, and `ShowLogo`. When you run Boilerplate, it will prompt
+the user (with the specified `prompt`, if specified) for each one.
+
+Next, create an `index.html` in the `website-boilerplate` folder that uses these variables using [Go
+Template](https://golang.org/pkg/text/template) syntax:
+
+```html
+{{snippet "../examples/website/index.html"}}
+```
+
+Copy an image into the `website-boilerplate` folder and call it `logo.png`.
+
+Finally, run `boilerplate`, setting the `--template-folder` to `website-boilerplate` and `--output-folder` to the path
+where you want the generated code to go:
+
+```
+boilerplate --template-folder /home/ubuntu/website-boilerplate --output-folder /home/ubuntu/website-output
+
+Enter the value for 'Title': Boilerplate Example
+Enter the welcome text for the website: Welcome!
+Should the website show the logo? (y/n) (default: "y"): y
+
+Generating /home/ubuntu/website-output/index.html
+Copying /home/ubuntu/website-output/logo.png
+```
+
+Boilerplate copies any files from the `--template-folder` into the `--output-folder`, passing them through the
+[Go Template](https://golang.org/pkg/text/template) engine along the way. After running the command above, the
+`website-output` folder will contain a `logo.png` (unchanged from the original) and an `index.html` with the
+following contents:
+
+```html
+{{snippet "../test-fixtures/examples-expected-output/website/index.html"}}
+```
+
+You can also run Boilerplate non-interactively, which is great for automation:
+
+```
+boilerplate \
+  --template-folder /home/ubuntu/website-boilerplate \
+  --output-folder /home/ubuntu/website-output \
+  --non-interactive \
+  --var Title="Boilerplate Example" \
+  --var WelcomeText="Welcome!" \
+  --var ShowLogo="y"
+
+Generating /home/ubuntu/website-output/index.html
+Copying /home/ubuntu/website-output/logo.png
+```
+
+Of course, Boilerplate can be used to generate any type of project, and not just HTML, so check out the
+[examples](/examples) folder for more examples and the [Working with Boilerplate](#working-with-boilerplate) section
+for full documentation.
+
+## Install
+
+Download the latest binary for your OS here: [boilerplate {{.Version}}](/releases/tag/{{.Version}}).
```

Comment:
At present, this syntax doesn't make sense to me but let's see what happens as I read futher.